tachyon/math/base/byinverter.h

```diff
@@ -0,0 +1,469 @@
+#ifndef TACHYON_MATH_BASE_BYINVERTER_H_
+#define TACHYON_MATH_BASE_BYINVERTER_H_
+
+namespace tachyon::math {
+/// Signed 64-bit integer type
+typedef __int64_t i64;
+/// Unsigned 64-bit integer type
+typedef __uint64_t u64;
+/// Unsigned 128-bit integer type
+typedef __uint128_t u128;
+/// Unsigned integer type for storing object sizes
+typedef size_t usize;
+
+/// Returns the non-largest argument
+constexpr usize least(usize a, usize b) { return a < b ? a : b; }
+
+/// Returns the number of trailing zeros in "n"
+/// represented in the little-endian binary code
+constexpr usize trailing0(u64 n) { return n != 0 ? __builtin_ctzll(n) : 64; }
+
+/// Big signed (B * L)-bit integer type, whose variables
+/// store numbers in the two's complement code as arrays
+/// of B-bit chunks, where B < 64 and L > 0. The ordering
+/// of the chunks in these arrays is little-endian. The
+/// arithmetic operations for this type are wrapping ones
+template <usize B, usize L> struct CInt {
```

Comment:
In most cases, `struct` uses for a container that has public member variables, so in this case, you should change it to `class`.

Reply:
Just a question, why do you name it to `CInt`? what does this stand for?

Reply:
Custom (or Chunked) Integer.